### PR TITLE
fix: validate wimse_domain at startup (SPIFFE §2.2 / RFC 1123)

### LINKED ISSUE #47: SPIFFE: trust domain format not validated at startup
## Summary

The SPIFFE specification (§2.2) requires that a trust domain name is a valid DNS name with only lower-case characters. ZeroID reads \`wimse_domain\` from configuration at startup but does not validate that the value is a well-formed, lower-case DNS name before using it as the trust domain component in all issued SPIFFE IDs.

## Location

Server startup / configuration loading — wherever \`wimse_domain\` (or equivalent) is read from env/config.

## Impact

An uppercase, whitespace-containing, or otherwise malformed trust domain is silently propagated into every SPIFFE ID the server issues, producing globally invalid identifiers.

## Fix

At startup (or config load time), validate that the trust domain:
1. Is a valid DNS hostname (RFC 1123).
2. Contains only lower-case characters.
3. Does not contain a leading \`spiffe://\` prefix (a common misconfiguration).

Fail fast with a clear error if validation fails.

## Reference

- [SPIFFE spec §2.2](https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE.md#22-trust-domain): "Trust domain names are defined by their corresponding authority … must be DNS names"

### PR DESCRIPTION
**Pull request by safayavatsal**
## Summary

- \`Config.Validate\` did not check \`wimse_domain\` at all, so an operator could set \`ZEROID_WIMSE_DOMAIN=Acme.Org\` or \`spiffe://example.com\` and the server would start cleanly, then mint SPIFFE IDs no SPIFFE library can parse.
- Adds \`validateWIMSEDomain\` covering the three failure modes the issue calls out — empty, leading \`spiffe://\`, non-lowercase / non-RFC-1123 — plus the standard RFC 1123 edge cases (label length 1–63, total length ≤253, no leading/trailing hyphens, no empty labels). Hooked from \`Config.Validate\` so it fails fast at startup.
- Each rejection names the offending field/label so the operator can fix the env var without reading the source.
- Table test covers 5 valid shapes + the 3 issue-required failures + 10 RFC 1123 edge cases.

Fixes #47

## Test plan

- [x] \`go vet ./...\` clean
- [x] \`go build ./...\` clean
- [x] \`go test .\` (root package, 18-case table)
- [x] \`make test\` passes (full integration suite, ~42s)

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -177,6 +177,48 @@ func (c *Config) Validate() error {
...
+		for i := 0; i < len(label); i++ {
+			c := label[i]
+			switch {
+			case c >= 'a' && c <= 'z':
+			case c >= '0' && c <= '9':
+			case c == '-':
+			default:
+				return fmt.Errorf("label %q contains character %q (allowed: a-z 0-9 -, lowercase only)", label, c)
+			}
+		}
```
[the remark this answers is not in the document]
Comment (safayavatsal) [reply config.go]: Good catch — fixed in c6787d1. Switched to `for _, r := range label` so a multi-byte UTF-8 char gets reported as the actual rune instead of its leading byte. Added a `café.example.com` test case to lock the behavior in.

[the remark this answers is not in the document]
Comment (safayavatsal) [reply config.go]: done

**Review by rsharath — APPROVED** [review]

**Review by rsharath — APPROVED** [review]
LGTM, approving.

Verified locally on this branch (now at `5884189`, merged-from-main):

- `go vet ./...` clean
- `go test -v -run TestValidateWIMSEDomain .` — 20/20 subtests pass (all four valid shapes, three explicit failure modes from the issue, and the RFC 1123 edge cases including the multi-byte rune case)
- Full integration suite green in ~10s

The validator catches the three failure modes named in #47 (empty / `spiffe://` prefix / uppercase) and the standard RFC 1123 cases (label length 1–63, total ≤253, no leading/trailing hyphens, no empty labels). Operator-actionable error messages — the substring assertions in the test pin them so the wording can't drift silently.
